_posts/2020-04-08-secure-software-development-lifecycle.md

```diff
@@ -0,0 +1,137 @@
+
+To sum up: 
+* Know the laws and regulations
+* Follow company policies for your own company and for your customers company
+* Check for useful standards
+* Make security requirements
+* Data classification
+
+#### Design
+
+Design sets a foundation for software architecture. Security was all about CIA (confidentiality, integrity and availability). Failures in software architecture most often affects one of the letters of CIA and they are also often pretty expensive to fix. To fight this we would like to setup extra care for design. There are plenty of tools and and practices in different software project management frameworks but I would like to bring out threat modeling. There are different approaches for threat modeling but what I have found useful is to create diagrams for software architecture from different approaches: deployment pipeline, infrastructure and networking, sequence diagrams and workflows. Reviewing those diagrams from the "how would I hack this?" -perspective is really useful. You can also go so far that you will create evil user stories for the backlog. 
+
+To sum up:
+* Create architecture diagrams and review them
+* Threat modeling 
+* Evil user stories
+
+
+#### Implementation
+
+Implementation is where the stuff that developers love happens. There are numerous ways to improve security or fail bad at this phase. Most important thing will be your developers security awareness. Many of the security issues are more or less stuff that nobody came to think about. Yes, I mean the OWASP Top 10 stuff like SQL injections. One thing you need to understand about developers is that they love to solve problems. So normal trick is to try to dress security requirements as a solvable problems like "make sure that all input is sanitized against sql injections" and "make sure that all output is sanitized against XSS". This is important but of course covers only one problem at a time and it will become a pretty exhaustive list to implement all over again in every project. Often times more eye-pairs for the same problem will help in finding loopholes in your security design. 
+
+As a software developer and security enthusiastic I could go on forever on this topic but more or less I will just put a list here things that you should consider:
+* Know or setup secure coding practices for your language and frameworks
+* Validate all the 3rd party libraries you take in
+* Fail securely
+* Make sure that your secrets management endures the daylight
```

Comment:
kestää päivänvaloa ei taida kääntyä näin suoraan englanniksi - ei nyt tule vastaavaa lausahdusta mieleen äkkiseltään englanniksi, joten ehkä parempi vain esittää jotenkin simppelisti